### PR TITLE
LobbyServer now creates Lobbies and send LobbyList

### PR DESCRIPTION
**Pull request by Deudly**
LobbyServer now creates Lobbies and send LobbyList

### CODE REVIEW

**Comment by Deudly** [comment]
Now Lobbies are accessible for clients and work perfect

File: server.js [hunk 1]
```diff
@@ -3,23 +3,26 @@ var io = require('socket.io')(port);
 var ClientManagerService = require('./app/services/ClientManagerService');
 var LobbyManagerService = require('./app/services/LobbyManagerService');
 
-console.log("League Sandbox Lobby Server");
 console.log("---------------------------");
+console.log("League Sandbox Lobby Server");
```
Comment (MythicManiac) [inline server.js:7]: In my opinion the old format for this worked more on par with most written content, as in

```
{title}
{section division}
{content}
```

now you have

```
{section division}
{title}
{content}
{section division}
```

but that's a pretty minor complaint, so no absolute need to change

File: lobby.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+"use strict";
+var port = 1234;
+const io = require('socket.io')(port);
+const repl = require('repl');
+
+const name = "My Lobby";
+const creator = "molen";
+const gamemode = "URF";
+
+const adminSettings = [{
```
Comment (MythicManiac) [inline lobby.js:12]: These settings should probably be in their own file

File: lobby.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+}, {
+    binding: "map",
+    name: "Map",
+    help: "Foo",
+    field: "mapSelect",
+    default: 11,
+    options: "*"
+}];
+
+const playerSettings = [{
```
Comment (MythicManiac) [inline lobby.js:31]: Ditto

File: app/services/ClientManagerService.js [hunk 4]
```diff
@@ -15,4 +15,4 @@ function ClientManagerService(){
   }
 }
 
-module.exports = ClientManagerService();
+module.exports = ClientManagerService();
```
Comment (MythicManiac) [inline app/services/ClientManagerService.js]: Text files should end in a newline, not sure why did you remove it

File: app/services/LobbyManagerService.js [hunk 5]
```diff
@@ -33,4 +34,4 @@ function LobbyManagerService(){
   }
 }
 
-module.exports = LobbyManagerService();
+module.exports = LobbyManagerService();
```
Comment (MythicManiac) [inline app/services/LobbyManagerService.js]: Ditto

File: app/services/LobbyManagerService.js [hunk 6]
```diff
@@ -7,6 +7,7 @@ function LobbyManagerService(){
   return {
     getLobbies: getLobbies,
     create: create,
+    getLobbyID: lobbyCount
```
Comment (MythicManiac) [inline app/services/LobbyManagerService.js]: This is far from reliable way to define a lobby Id.
Also, following the set naming convention, this member should be called `getLobbyId`

File: app/factories/LobbyFactory.js [hunk 7]
```diff
@@ -8,11 +9,24 @@ function LobbyFactory(){
     var lobby;
 
     lobby = {
-      name: options.name || "",
-      hostSettings: options.hostSettings || [],
-      playerSettings: options.playerSettings || []
+      //Just for test we disable this
+      //name: options.name || "",
+      //hostSettings: options.hostSettings || [],
```
Comment (MythicManiac) [inline app/factories/LobbyFactory.js]: You should either remove these or leave them uncommented. I'm guessing you commented these lines because you didn't know how they work, which isn't really a valid solution.

**Comment by MythicManiac** [comment]
Overall this still seems very much like a WIP, and I'm assuming `lobby.js` is copied from the LobbyClient?

I'll merge this once the issues I pointed out are addressed

File: app/services/ClientManagerService.js (hunk 4 above)
Comment (Deudly) [inline app/services/ClientManagerService.js]: It is 100% identic, WHY IS THAT A PROBLEM? JUST MISSCLICK

File: app/services/LobbyManagerService.js (hunk 5 above)
Comment (Deudly) [inline app/services/LobbyManagerService.js]: and that is a problem @MythicManiac ?missclick

File: app/services/LobbyManagerService.js (hunk 6 above)
Comment (Deudly) [inline app/services/LobbyManagerService.js]: ok

File: app/factories/LobbyFactory.js (hunk 7 above)
Comment (DeudlyYT) [inline app/factories/LobbyFactory.js]: I know how they work but I commented them because before I start working on them I enabled a options which already are configured XD @MythicManiac

Comment (Deudly) [inline app/factories/LobbyFactory.js]: I know how they work but I commented them because before I start working on them I enabled a options which already are configured XD @MythicManiac Also they didnt work and I'll solve later

File: lobby.js (hunk 3 above)
Comment (Deudly) [inline lobby.js:31]: @MythicManiac I dont understand this so I wont edit nothing until I understand all. Also this is not revelant

**Comment by MythicManiac** [comment]
I'm still not entirely sold by the lobby id / options shenanigans, but I guess this'll do for now
